market/connector/docusign/demo/eSignatureDemoTest/src_test/net/docusign/test/TestDocuSignDemo.java

```diff
@@ -7,22 +7,32 @@
 import ch.ivyteam.ivy.bpm.engine.client.BpmClient;
 import ch.ivyteam.ivy.bpm.engine.client.ExecutionResult;
 import ch.ivyteam.ivy.bpm.exec.client.IvyProcessTest;
+import ch.ivyteam.ivy.environment.AppFixture;
 import ch.ivyteam.ivy.security.ISession;
 
 @IvyProcessTest
 public class TestDocuSignDemo
 {
   @Test
-  public void main(BpmClient bpmClient, ISession session)
+  public void main(BpmClient bpmClient, ISession session, AppFixture fixture)
   {
-    session.setActiveEnvironment("dev-axonivy");
+    fixture.environment("dev-axonivy");
     
-    ExecutionResult result2 = bpmClient.start()
-      .process("Demo/eSign/authenticated.ivp")
+    ExecutionResult result = bpmClient.start()
+      .process("Demo/eSign/start.ivp")
       .as().session(session)
-      .withParam("code", "a-test-code")
       .execute();
-    
+    assertThat(result.http().redirectLocation()).containsSubsequence(
+            "http://localhost:",
+            "/test/api/docuSignMock/oauth/auth?response_type=code&scope=signature+impersonation&client_id=test-key&redirect_uri=http%3A%2F%2Flocalhost%3A", 
+            "%2Ftest%2Fauth%2Fcallback");
+
+    ExecutionResult result2 = bpmClient.start()
+            .webPage(result.workflow().executedTask(), "17651A0FB6AFB366/17651A0FB6AFB366-f6/resume.ivp")
```

Comment:
looks as if we should introduce new explicit API to cover this use-case ...what's your opinion? should we create a story for it?

Reply:
That was also my first idea. 
But on a second though: how many of our customers will try to test this redirection? 
Most of them will simply mock the whole REST call when testing the process.
So, introducing a new API point for a use case that is seldom used will enhance API complexity and the effort is not worth it.
Let's wait and see  if a customer comes with a feature request for this. If customers need them we can introduce an own explicit API.